basyx/aas/adapter/http.py

```diff
@@ -759,8 +790,9 @@ def aas_submodel_refs_redirect(self, request: Request, url_args: Dict, map_adapt
     # ------ SUBMODEL REPO ROUTES -------
     def get_submodel_all(self, request: Request, url_args: Dict, **_kwargs) -> Response:
         response_t = get_response_type(request)
-        submodels = self._get_submodels(request)
-        return response_t(list(submodels), stripped=is_stripped_request(request))
+        submodels = self._get_submodels(request)[0]
+        cursor = self._get_submodels(request)[1]
```

Comment:
This is incredibly inefficient because you're calling `_get_submodels()` twice, effectively collecting and slicing all submodels two times. You can use the following shorthand instead:
```python
submodels, cursor = self._get_submodels(request)
```